nixos/modules/services/continuous-integration/concourse/worker.nix

```
@@ -0,0 +1,224 @@
+          "local-fs.target"
+          "dbus.service"
+        ];
+        wantedBy = [ "multi-user.target" ];
+        path = [
+          pkgs.util-linux
+          pkgs.iptables
+          pkgs.gnutar
+          pkgs.gzip
+          pkgs.runc
+        ]
+        # From `containerd`
+        ++ lib.optional config.boot.zfs.enabled config.boot.zfs.package;
+        serviceConfig = {
+          # Worker must be run as root, because it needs to launch containers
+          #WorkingDirectory = cfg.workDir;
+          StateDirectory = cfg.workDir;
+          StateDirectoryMode = "0777";
+          #UMask = "0007";
+          ConfigurationDirectory = "concourse-worker";
+          EnvironmentFile = cfg.environmentFile;
+          ExecStart = "${cfg.package.override { withWebUI = false; }}/bin/concourse worker ${cfg.args}";
+          Restart = if cfg.auto-restart then "on-failure" else "no";
+          RestartSec = 15;
+
+          # From containerd
+          Delegate = "yes";
+          LimitNPROC = "infinity";
+          LimitCORE = "infinity";
+          TasksMax = "infinity";
+          OOMScoreAdjust = "-999";
+          #CapabilityBoundingSet = "";
+          # Security
+          #NoNewPrivileges = true;
+          ## Sandboxing
+          #ProtectSystem = "full";
+          #ProtectHome = true;
+          #PrivateTmp = true;
+          #PrivateDevices = true;
+          #PrivateUsers = true;
+          #ProtectHostname = true;
+          #ProtectClock = true;
+          #ProtectKernelTunables = true;
+          #ProtectKernelModules = true;
+          #ProtectKernelLogs = true;
+          #LockPersonality = true;
+          #MemoryDenyWriteExecute = true;
+          #RestrictRealtime = true;
+          #RestrictSUIDSGID = true;
+          #PrivateMounts = true;
+
+          #RestrictAddressFamilies = [ "AF_UNIX AF_INET AF_INET6" ];
+
+          # Do not filter control groups and system calls since this needs to run a container runtime
+          #ProtectControlGroups = true;
+          #SystemCallArchitectures = "native";
+          #SystemCallFilter = "~@clock @privileged @cpu-emulation @debug @keyring @module @mount @obsolete @raw-io @reboot @setuid @swap";
+        };
+        environment = {
+          CONCOURSE_WORK_DIR = cfg.workDir;
+          CONCOURSE_TAG = cfg.tag;
+          CONCOURSE_TEAM = cfg.team;
+
+          CONCOURSE_TSA_HOST = cfg.tsa.host;
+          CONCOURSE_TSA_PUBLIC_KEY = cfg.tsa.publicKey;
+          CONCOURSE_TSA_WORKER_PRIVATE_KEY = cfg.tsa.workerPrivateKey;
+
+          CONCOURSE_BAGGAGECLAIM_BIND_IP = cfg.p2p.bindIP;
+          CONCOURSE_BAGGAGECLAIM_P2P_INTERFACE_NAME_PATTERN = cfg.p2p.interfaceNamePattern;
+          CONCOURSE_BAGGAGECLAIM_P2P_INTERFACE_FAMILY = cfg.p2p.interfaceFamily;
+
+          CONCOURSE_RUNTIME = cfg.runtime.type;
+          CONCOURSE_RESOURCE_TYPES = lib.defaultTo "${pkgs.concourse.resource-types}" cfg.resourceTypes;
+        }
+        // lib.ifEnable useContainerd {
```

ifEnable is deprecated. 

```suggestion
        // lib.optionalAttrs useContainerd {
```